nemo/collections/asr/parts/utils/diarization_utils.py

```diff
@@ -925,80 +898,147 @@ def write_result_in_csv(self, args, WDER_dict, DER_result_dict, effective_WDER):
+        logging.info(f"Writing {target_path}")
+        csv_columns = [
+            'uniq_id',
+            'DER',
+            'CER',
+            'FA',
+            'MISS',
+            'est_n_spk',
+            'is_spk_count_correct',
+            'ref_word_count',
+            'hyp_word_count',
+            'diar_confuse_count',
+            'all_correct_count',
+            'diar_trans_acc',
+            'hyp_based_wder',
+            'ref_based_wder',
+            'rttm_based_wder',
+            'mapping',
+        ]
+        dict_data = [x for k, x in WDER_dict['session_level'].items()]
+        try:
+            with open(target_path, 'w') as csvfile:
+                writer = csv.DictWriter(csvfile, fieldnames=csv_columns)
+                writer.writeheader()
+                for data in dict_data:
+                    writer.writerow(data)
+        except IOError:
+            logging.info("I/O error has occurred while writing a csv file.")
+
+    def break_lines(self, string_out, max_chars_in_line=90):
+        """
+        Break the lines in the transcript.
+        """
+        color_str_len = len('\033[1;00m') if self.params['colored_text'] else 0
+        split_string_out = string_out.split('\n')
+        return_string_out = []
+        for org_chunk in split_string_out:
+            buffer = []
+            if len(org_chunk) - color_str_len > max_chars_in_line:
+                color_str = org_chunk[:color_str_len] if color_str_len > 0 else ''
+                for i in range(color_str_len, len(org_chunk), max_chars_in_line):
+                    trans_str = org_chunk[i : i + max_chars_in_line]
+                    if len(trans_str.strip()) > 0:
+                        c_trans_str = color_str + trans_str
+                        buffer.append(c_trans_str)
+                return_string_out.extend(buffer)
+            else:
+                return_string_out.append(org_chunk)
+        return '\n'.join(return_string_out)
 
     def write_and_log(self, uniq_id, riva_dict, string_out, audacity_label_words, gecko_dict):
-        """Writes output files and display logging messages.
+        """
+        Write output files and displays logging messages.
         """
         ROOT = self.root_path
+
         logging.info(f"Writing files for id:{uniq_id} at {ROOT}/pred_rttms/")
+
+        logging.info(f"Writing {ROOT}/pred_rttms/{uniq_id}.json")
         dump_json_to_file(f'{ROOT}/pred_rttms/{uniq_id}.json', riva_dict)
+
+        logging.info(f"Writing {ROOT}/pred_rttms/{uniq_id}_gecko.json")
         dump_json_to_file(f'{ROOT}/pred_rttms/{uniq_id}_gecko.json', gecko_dict)
+
+        logging.info(f"Writing {ROOT}/pred_rttms/{uniq_id}.txt")
         write_txt(f'{ROOT}/pred_rttms/{uniq_id}.txt', string_out.strip())
+
+        logging.info(f"Writing {ROOT}/pred_rttms/{uniq_id}.w.label")
         write_txt(f'{ROOT}/pred_rttms/{uniq_id}.w.label', '\n'.join(audacity_label_words))
```

Comment:
Can we not write info for each uniq_id, just info ing in which directory these are being stored would be enough, else sys out is becoming total cluttered.

Reply:
It is reduced to one line.